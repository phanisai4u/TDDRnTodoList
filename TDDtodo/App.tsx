import React from 'react';
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  Button,
} from 'react-native';


export interface MyChildComProps {
  handleButtonPress?:()=> void,
  text?:string
};

export class MyChildCom extends React.Component<MyChildComProps> {
 constructor(props: MyChildComProps) {
    super(props);
    this.state = {
      text: props.text || ""
    }
  }

 
  render() {
    return (
      <View style={styles.appContent}>
        <Text>{this.state.text}</Text>
        <Button testID ={"helloiii"} title={"Press me"} onPress={this.props.handleButtonPress}> </Button>
      </View>

    )
  }
}


export const add = (a: number, b: number) => a + b;



export class App extends React.Component {

  constructor(props: any) {
    super(props);
    this.state = {
      test: props.test || false,
      text: props.text || ""
    }
  }

  internalAdd(a: number, b: number) {
    return a + b;
  }

  componentDidMount() {
    this.setState({ test: false })
  }

  changeTest(val: boolean) {
    this.setState({ test: val })
  }
  handleTextChange = (val: string) => {
    this.setState({ text: val })
  }
  handleButtonPress = () => {
    this.setState({ text: "ButtonPressed" })
  }
  stateChangeFromChild = () => {
    this.setState({ text: "ChildButtonPressed" })

  }
  render() {
    const test: boolean = this.state.test || false
    return (
      test ? <View style={styles.appContent}>
        <Text> Test is please case</Text>
      </View> : <View style={styles.appContent}>
          <Text> hi</Text>
          <TextInput onChangeText={this.handleTextChange} value={this.state.text}></TextInput>
          <TextInput onChangeText={this.handleTextChange} value={this.state.text}></TextInput>

          <Button title={"Press me"} onPress={this.handleButtonPress}> </Button>
          <MyChildCom text = {"newChild"} handleButtonPress={this.stateChangeFromChild} />
        </View>

    )

  }
};



const styles = StyleSheet.create({
  appContent: {
    backgroundColor: 'gray',
  }
});

export default App;
