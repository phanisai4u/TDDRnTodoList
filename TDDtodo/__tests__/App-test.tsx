/**
 * @format
 */


 import 'react-native';

import React from 'react';

import { shallow } from 'enzyme';

import App,{add, MyChildCom} from '../App';
import { TextInput, Button } from 'react-native';


const createTestProps = (props: object) => ({
  ...props,
  test:false,
  text:''
});

describe('App', () => {
  const props = createTestProps({});
  const wrapper = shallow<App>(<App {...props} />);
  const Child = shallow<App>(<MyChildCom text={"Child"} />);
  const Child2 = shallow<App>(<MyChildCom  />);



  describe('rendering', () => {

    it('Shallow rendering', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('Child Shallow rendering', () => {
      expect(Child).toMatchSnapshot()
    })

    it('Child for else Shallow rendering', () => {
      expect(Child2).toMatchSnapshot()
    })


    it('should render a <View />', () => {
      expect(wrapper.find('View')).toHaveLength(1);
    });

    // it('should add two numbers', () => {
    //   expect(add(1, 1)).toEqual(2);
    // });

    //Test class internal functions
    it('should add two numbers', () => {
      //NOTE:- Here wrapper must be React element other wise instance() function  will return null
      expect(wrapper.instance().internalAdd(1, 1)).toEqual(2);
    });
    
    // it('has state', () => {
    //   expect(wrapper.state('test')).toBe(false);
    // });

    // //For negative test case for failure
    // it('has state', () => {
    //   expect(wrapper.state('test')).toBe(true);
    // });

    //Update here  
    //after update this TestCase will be true
    it ('update state',() => {
      wrapper.instance().changeTest(true)
      expect(wrapper.state('test')).toBe(true);
    })

    // it ('update test input text',() => {
    //   wrapper.instance().changeTest(false)


    //   expect(wrapper.find('TextInput')).toHaveLength(2);
    //   wrapper.find('TextInput').at(0).simulate('changeText','Hello')
    //   expect(wrapper.state('text')).toBe('Hello');
    // })

    // it ('update test Button Press',() => {
    //   wrapper.instance().changeTest(false)
    //   wrapper.find(Button).at(0).simulate('press')
    //   expect(wrapper.state('text')).toBe('ButtonPressed');
    // })


    // it ('update Child component events',() => {
    //   wrapper.instance().changeTest(false)
    //   expect(wrapper.find('MyChildCom')).toHaveLength(1);
    //   console.log( wrapper.find('MyChildCom').at(0).debug())
    //   wrapper.find('MyChildCom').at(0).props().handleButtonPress()
    //   expect(wrapper.state('text')).toBe('ChildButtonPressed');
    // })

  });
});


