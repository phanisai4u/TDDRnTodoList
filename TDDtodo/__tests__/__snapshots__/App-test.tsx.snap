// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering Child Shallow rendering 1`] = `ShallowWrapper {}`;

exports[`App rendering Child for else Shallow rendering 1`] = `ShallowWrapper {}`;

exports[`App rendering Shallow rendering 1`] = `ShallowWrapper {}`;
